### PR TITLE
feat: Milvus bump up to 2.4

### LINKED ISSUE #481: [Feature Request] Milvus bump up to 2.4
### Required prerequisites

- [X] I have searched the [Issue Tracker](https://github.com/camel-ai/camel/issues) and [Discussions](https://github.com/camel-ai/camel/discussions) that this hasn't already been reported. (+1 or comment there if it has.)
- [X] Consider asking first in a [Discussion](https://github.com/camel-ai/camel/discussions/new).

### Motivation

milvus launched 2.4 version in yesterday
it enables Multi-vector Search for Simplified Multimodal Searches and also added support for re-rank model
these updates is very useful for our current RAG design

https://milvus.io/blog/milvus-2-4-nvidia-cagra-gpu-index-multivector-search-sparse-vector-support.md

### Solution

_No response_

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Wendong-Fan**
## Description

milvus launched 2.4 version in yesterday
it enables Multi-vector Search for Simplified Multimodal Searches and also added support for re-rank model
these updates is very useful for our current RAG design

https://milvus.io/blog/milvus-2-4-nvidia-cagra-gpu-index-multivector-search-sparse-vector-support.md

## Motivation and Context

`close #481` 

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by Wendong-Fan** [comment]
lucky me, I merged the old version of Milvus right on the day they released a shiny new one version...

File: camel/storages/vectordb_storages/milvus.py [hunk 1]
```diff
@@ -118,40 +118,46 @@ def _create_collection(
...
-                dim=self.vector_dim)
-        ]
-        schema = CollectionSchema(fields=fields,
-                                  description='Collection schema')
-
-        self._client.create_collection_with_schema(
+        from pymilvus import DataType
+
+        # set schema
+        schema = self._client.create_schema(auto_id=False,
```
Comment (Appointat) [inline camel/storages/vectordb_storages/milvus.py:128]:
```suggestion
        # Set the schema
        schema = self._client.create_schema(auto_id=False,
```

File: camel/storages/vectordb_storages/milvus.py [hunk 2]
```diff
@@ -118,40 +118,46 @@ def _create_collection(
...
+            field_name="vector", datatype=DataType.FLOAT_VECTOR,
+            description='The numerical representation of the vector',
+            dim=self.vector_dim)
+        schema.add_field(
+            field_name="payload", datatype=DataType.JSON,
+            description=('Any additional metadata or information related'
+                         'to the vector'))
+
+        # create collection
+        self._client.create_collection(
```
Comment (Appointat) [inline camel/storages/vectordb_storages/milvus.py:145]:
```suggestion
        # Create the collection
        self._client.create_collection(
```

File: camel/storages/vectordb_storages/milvus.py [hunk 3]
```diff
@@ -118,40 +118,46 @@ def _create_collection(
...
             collection_name=collection_name,
             schema=schema,
-            index_params=index_params,
             **kwargs,
         )
 
+        # set index
```
Comment (Appointat) [inline camel/storages/vectordb_storages/milvus.py:151]:
```suggestion
        # Set the index of the parameters
```

File: camel/storages/vectordb_storages/milvus.py [hunk 4]
```diff
@@ -277,7 +281,7 @@ def delete(
         Args:
             ids (List[str]): List of unique identifiers for the vectors to be
                 deleted.
-            **kwargs (Any): Additional keyword arguments.
+            **kwargs (Any): Additional keyword arguments pass to delete.
```
Comment (Appointat) [inline camel/storages/vectordb_storages/milvus.py:285]:
```suggestion
            **kwargs (Any): Additional keyword arguments passed to delete.

```

File: camel/storages/vectordb_storages/milvus.py [hunk 5]
```diff
@@ -313,7 +317,7 @@ def query(
         Args:
             query (VectorDBQuery): The query object containing the search
                 vector and the number of top similar vectors to retrieve.
-            **kwargs (Any): Additional keyword arguments.
+            **kwargs (Any): Additional keyword arguments pass to search.
```
Comment (Appointat) [inline camel/storages/vectordb_storages/milvus.py:321]:
```suggestion
            **kwargs (Any): Additional keyword arguments passed to search.

```

File: camel/storages/vectordb_storages/milvus.py [hunk 6]
```diff
@@ -118,40 +118,46 @@ def _create_collection(
...
+
+        # set schema
+        schema = self._client.create_schema(auto_id=False,
+                                            enable_dynamic_field=True,
+                                            description='collection schema')
+
+        schema.add_field(field_name="id", datatype=DataType.VARCHAR,
+                         descrition='A unique identifier for the vector',
+                         is_primary=True, max_length=65535)
+        schema.add_field(
```
Comment (Appointat) [inline camel/storages/vectordb_storages/milvus.py:135]: Do you think it's necessary to provide the ref for the number '65535'? (`https://milvus.io/docs/limitations.md`)

**Review by Appointat — APPROVED** [review]
Good updates, and I left some comments for you.

File: camel/storages/vectordb_storages/milvus.py (hunk 6 above)
Comment (Wendong-Fan) [reply camel/storages/vectordb_storages/milvus.py]: thanks for the review, added reference

**Comment by Wendong-Fan** [comment]
Haven't update all new capabilities yet since it's still in milvus's beta version , will do it later when we need further multi-modal support

**Pull request merged**